### PR TITLE
tkt-34591: Require passphrase to lock pool

### PR DESCRIPTION
**Pull request by sonicaj**
This commit ensures that a passphrase is required when locking a pool to prevent unauthorized personnel accidentally locking the pool.
Ticket: #34591

### CODE REVIEW

File: gui/api/resources.py [hunk 1]
```diff
@@ -958,15 +960,30 @@ def lock(self, request, **kwargs):
 
         bundle, obj = self._get_parent(request, kwargs)
 
-        if obj.vol_encrypt == 0:
-            raise ImmediateHttpResponse(
-                response=self.error_response(request, _('Volume is not encrypted.'))
-            )
```
Comment (william-gr) [inline gui/api/resources.py]: Why was this removed?

File: src/middlewared/middlewared/plugins/pool.py [hunk 2]
```diff
@@ -1552,9 +1552,9 @@ def __find_disk_from_topology(self, label, pool):
         return True
 
     @item_method
-    @accepts(Int('id'))
+    @accepts(Int('id'), Str('passphrase'))
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py]: password=True

File: gui/api/resources.py (hunk 1 above)
Comment (sonicaj) [reply gui/api/resources.py]: So we already raise a validation error when locking a pool if that pool is not encrypted. This was redundant in that case
